### PR TITLE
Update xcode to 14e222b

### PR DESCRIPTION
**Pull request by cyanglaz**
Framework part of https://github.com/flutter/flutter/issues/125814

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cyanglaz** [comment]
Mac plugin_lint_mac failed:
```
[2023-04-28 14:59:00.797801] [STDOUT] stdout:  -> integration_test (0.0.1)
[2023-04-28 14:59:00.797814] [STDOUT] stdout:     - NOTE  | xcodebuild:  note: Using codesigning identity override: -
[2023-04-28 14:59:00.797827] [STDOUT] stdout:     - NOTE  | [iOS] xcodebuild:  note: Building targets in dependency order
[2023-04-28 14:59:00.797840] [STDOUT] stdout:     - WARN  | xcodebuild:  /Volumes/Work/s/w/ir/x/t/CocoaPods-Lint-20230428-66011-7pgvqe-integration_test/App/main.m:4:9: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
[2023-04-28 14:59:00.797854] [STDOUT] stdout:     - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'Flutter' from project 'Pods')
[2023-04-28 14:59:00.797868] [STDOUT] stdout: 
[2023-04-28 14:59:00.797880] [STDOUT] stdout: [!] integration_test did not pass validation, due to 1 warning (but you can use `--allow-warnings` to ignore it).
[2023-04-28 14:59:00.797893] [STDOUT] stdout: You can use the `--no-clean` option to inspect any issue.
```

**Comment by cyanglaz** [comment]
Trying to run `bin/cache/dart-sdk/bin/dart --disable-dart-dev --enable-vm-service=0 --no-pause-isolates-on-exit bin/tasks/plugin_lint_mac.dar` locally but it hang and I got:
```
The Dart VM service is listening on http://127.0.0.1:51334/uZK7dl_Ajvg=/
SEVERE: 2023-04-28 15:21:55.069131: Task did not start in 0:01:00.000000.
```

File: .ci.yaml [hunk 1]
```diff
@@ -78,7 +78,7 @@ platform_properties:
         ]
       os: Mac-12
       device_type: none
-      xcode: 14c18
+      xcode: 14e222b
```
Comment (hellohuanlin) [inline .ci.yaml:81]: wondering if it's better to use a variable so next time we only need to update one place?

**Comment by jmagman** [comment]
> Mac plugin_lint_mac failed:

Filed https://github.com/flutter/flutter/issues/125812

**Comment by jmagman** [comment]
> Mac plugin_lint_mac failed:

In the meantime we can add `--allow-warnings` with a TODO to remove when infra updates the necessary gem in these two spots:
https://github.com/flutter/flutter/blob/db7196c52cf8ce7313dc61139f095d5d29615e48/dev/devicelab/bin/tasks/plugin_lint_mac.dart#L41
https://github.com/flutter/flutter/blob/db7196c52cf8ce7313dc61139f095d5d29615e48/dev/devicelab/bin/tasks/plugin_lint_mac.dart#L52

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**